### PR TITLE
Fix migration back and forth local event handler

### PR DESCRIPTION
**Pull request by xwduan**
- Fix migration back and forth usecase and add more tests
- fix lint

## What changed?

1. Fix a bug on handling replication back and forth case
2. Add more integration test to test the ^ scenario
## Why?

LocalGeneratedEventHandler cannot properly handle force replication case.
## How did you test it?

Integration test
## Potential risks

Feature is gated by a feature flag
## Documentation

n/a
## Is hotfix candidate?

no.

### CODE REVIEW

File: service/history/replication/eventhandler/local_events_handler.go [hunk 1]
```diff
@@ -226,6 +259,18 @@ func (h *localEventsHandlerImpl) importEvents(
 	return nil
 }
 
+func (h *localEventsHandlerImpl) isLastEventAtHistoryBoundary(
+	lastLocalEvent *historypb.HistoryEvent,
+	versionHistory *historyspb.VersionHistory,
+) bool {
+	for _, item := range versionHistory.Items {
```
Comment (yux0) [inline service/history/replication/eventhandler/local_events_handler.go:266]: nit: versionhistory.GetLastVersionHistoryItem(

File: service/history/replication/eventhandler/local_events_handler.go [hunk 2]
```diff
@@ -115,6 +116,29 @@ func (h *localEventsHandlerImpl) HandleLocalGeneratedHistoryEvents(
...
+
+	switch err.(type) {
+	// if mutable state not found, we import from beginning
+	case *serviceerror.NotFound:
+		return h.importEvents(
+			ctx,
+			sourceClusterName,
+			engine,
+			workflowKey,
+			1,
```
Comment (yux0) [inline service/history/replication/eventhandler/local_events_handler.go:135]: is there a common.FirstEventID?

File: service/history/replication/eventhandler/local_events_handler.go [hunk 3]
```diff
@@ -194,7 +220,14 @@ func (h *localEventsHandlerImpl) importEvents(
...
 		blobs = append(blobs, batch.RawEventBatch)
-		if blobSize >= historyImportBlobSize || len(blobs) >= historyImportPageSize {
+		events, err := h.eventSerializer.DeserializeEvents(batch.RawEventBatch)
+		if err != nil {
+			return err
+		}
+
+		if blobSize >= historyImportBlobSize ||
+			len(blobs) >= historyImportPageSize ||
+			h.isLastEventAtHistoryBoundary(events[len(events)-1], versionHistory) { // Import API only take events that has same version
```
Comment (yux0) [inline service/history/replication/eventhandler/local_events_handler.go:230]: where do we check the events must have same version?

Comment (xwduan) [reply service/history/replication/eventhandler/local_events_handler.go:230]: It's here: https://github.com/temporalio/temporal/blob/00fab090ff656dcff2a9e669e7835cdd6e893dbf/service/history/ndc/replication_task.go#L518

**Review by yux0 — APPROVED** [review]

**Pull request merged**